monix-tail/shared/src/main/scala/monix/tail/internal/IterantFoldLeft.scala

```diff
@@ -31,37 +33,51 @@ private[tail] object IterantFoldLeft {
   final def apply[F[_], S, A](source: Iterant[F, A], seed: => S)(op: (S,A) => S)
     (implicit F: Sync[F]): F[S] = {
 
-    def loop(self: Iterant[F, A], state: S): F[S] = {
+    def loop(stopRef: ObjectRef[F[Unit]])(self: Iterant[F, A], state: S): F[S] = {
```

Comment:
Lets do currying here. Put the `state` on the left, such that you can then do ... 

```scala
rest.flatMap(loop(stopRef, newState))
```